run-csharp

```diff
@@ -2,10 +2,71 @@
 
 . ./config
 
+CSHARP_TEST_FORMATS_DIR=$FORMATS_COMPILED_DIR/csharp
 CSHARP_SPEC_DIR=spec/csharp
+CSHARP_TEST_OUT_DIR="$TEST_OUT_DIR"/csharp
 
-# Compile everything
-xbuild "$CSHARP_SPEC_DIR/kaitai_struct_csharp_tests.sln"
+rm -rf "$CSHARP_TEST_OUT_DIR"
+mkdir -p "$CSHARP_TEST_OUT_DIR"
+
+# Try and build the project first - if there's no compile errors, we can continue straight away
+echo "run-csharp: building project"
+xbuild "$CSHARP_SPEC_DIR/kaitai_struct_csharp_tests.sln" >> "$CSHARP_TEST_OUT_DIR/_compile.log" || {
```

Comment:
_compile.log is always appended to here, never cleared. Probably it's a good idea to do `rm -f "$CSHARP_TEST_OUT_DIR/_compile.log"` somewhere at the beginning? Also, if `"$CSHARP_TEST_OUT_DIR/_compile.log"` is used more than once, probably it's worth to make a new variable for it?